### PR TITLE
Better db seeding 462 part2

### PR DESCRIPTION
**Pull request by Jagjeet**
This PR does the following relating to issue #462 :

-Restructures the seeding scripts to separate CLI commands from the database seeding
-Adds a script for creating message threads with the following usage: node fillTestMessageData <number of threads to add> <max messages per thread>
-Moves the previous 'User' functionality into a script called 'fillTestUserData.js'
-Combines the logic for creating users and admins into a single function
-Moves the individual seeding scripts into a 'db-seeding' directory
-Creates a master 'fillUserData.js' script that is preconfigured with values that should hopefully be good enough for most cases.

Further issues will be addressed more individually to avoid giant PRs. These will potentially include:
-Adding mock messaging between admin users
-Adding mock images to Tribes
-Improving randomization of message times and read/unread flags
-Soliciting feedback to see what else is valuable to make the mock data more useful. (eg. We could separate out user creation and offer creation, but if nobody needs it there is no point)

In any case, hopefully it will be easier for me or others to make smaller changes individually now that the code is slightly reorged. Let me know if there are any questions.

### CODE REVIEW

File: scripts/db-seeding/fillTestMessagesData.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+      }).catch(function (err) {
+        console.log(err);
+      });
+    });
+  });
+};
+
+// Number of threads and max number of messages is required
+if (process.argv[2] == null || process.argv[2] < 1 || process.argv[3]== null || process.argv[3] < 1) {
```
Comment (simison) [inline scripts/db-seeding/fillTestMessagesData.js]: As a future improvement, we could build this stuff using [Yargs](http://yargs.js.org/), but that's for another PR.

**Comment by simison** [comment]
What do you think if we'd move these scripts to `/bin` folder and access them via `npm` scripts configured in [`package.json`](https://github.com/Trustroots/trustroots/blob/9574266d8d1a04bde96f1093b12e208af2205126/package.json#L23-L55)?

For example:
```
npm run test-data
```
```
npm run test-data:tribes 20
```
would run:
```json
{
  "test-data": "npm run test-data:*",
  "test-data:tribes": "node ./bin/fillTestData/tribes.js",
  "test-data:messages": "node ./bin/fillTestData/messages.js",
}
```

...or something like that.

**Comment by Jagjeet** [comment]
That sounds good! It's a quick change and I am happy to do it as part of this request. Coming from outside the JS world, I tend to forget to centralize things through package.json, but it is definitely better that way. :)

**Comment by Jagjeet** [comment]
This is all ready to go now. Going to address adding images to the tribes as another PR.

**Comment by simison** [comment]
Works like a charm 👌 

As a future iteration, we could allow setting Mongoose's debug -mode off during the script's run because now the console output is quite jarring.

https://github.com/Trustroots/trustroots/blob/bf775bbb08dd65f3cea2ea1b924868e6c4333d1e/config/env/development.js#L23-L24

Environment variables could work for that.

**Pull request merged**